### ISSUE TITLE
Redundant error displayed on successful orders if previous attempt failed

### ISSUE THREAD

**Issue by timur27** [Automattic/woocommerce-payments#6141]
### Describe the bug
Users see a non-true error while successfully checking out on the shortcode checkout with a payment method that requires bank authentication (e.g. giropay), only when this order previously failed. 

### To Reproduce

1. Enable the split UPE in WCPay Dev Settings
2. Buy any product with the shortcode checkout, use e.g. giropay, and on the bank authentication page - fail the transaction. 
3. After being redirected back to the checkout page, _Place order_ once again and authenticate the transaction this time. 
4. After being redirected to the successful order details page, the error `Unfortunately your order cannot be processed as the originating bank/merchant has declined your transaction. Please attempt your purchase again.` is displayed. At the same time, the order is successfully processed and has the right status.

### Screen recording
https://user-images.githubusercontent.com/22319444/234223883-ae8e73b8-9611-4eea-8756-477af8910f95.mov

### Expected behavior
For split UPE and deferred intent creation UPE, users don't see the error on the order details page after successfully placing an order. Even in case, this order failed previously.

### Additional context
* The root cause of this issue is that the order is still in the status failed when performing a successful checkout (webhook from Stripe doesn't arrive yet), and [this WC Core implementation](https://github.com/Automattic/woocommerce.com/blob/trunk/plugins/woocommerce/templates/checkout/thankyou.php?rgh-link-date=2023-04-17T12%3A39%3A03Z#L29-L31) is executed.
* The legacy UPE doesn't have this problem. Analyzing it as part of the defect resolving might be a good step to start with.

**Comment by timur27** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Heisenberg-63586f2ec60359001310fa6a/issues/Automattic/woocommerce-payments/6141?planning-poker) with Zenhub @FangedParakeet @mdmoore

**Comment by mdmoore** [comment]
It looks like this will be fixed by #6170 but it's worth another test after that PR is merged.

**Comment by mdmoore** [comment]
Confirming that this issue is no longer present after #6170 was merged.

https://github.com/Automattic/woocommerce-payments/assets/3473953/5ea82ce0-51ca-49f5-a487-3ef7064fd4f1


<img width="282" alt="Screen Shot on 2023-05-10 at 16-42-03(1)" src="https://github.com/Automattic/woocommerce-payments/assets/3473953/b354a24f-e31a-40d2-b299-fe30b46ac655">
